docs: Enterprise Use Case Tutorial - User Management, Access Controls, and Auditing with Feature Flags (#8998)
Unleash/unleash: website/docs/feature-flag-tutorials/use-cases/user-management-access-controls.md

This automation creates an access control system that adapts in real-time to organizational changes. Within minutes of joining, a new team member can be granted precisely the right level of access, while departing employees are immediately locked out of sensitive systems.

## Configure role-based access controls

[Role-based access control](/reference/rbac) makes feature flag management a strategic governance tool. Consider a scenario in which a junior developer should never be able to modify critical feature flags in the authentication system within a banking platform. RBAC makes this granular control seamless. We recommend carefully delegating administrative privileges to users based on the needs of their roles within projects and the organization.

In the Unleash Admin UI, go to **Admin > Roles** to view, create, and manage user roles.

![Manage all user roles in the Unleash Admin UI.](/img/use-case-user-mgmt-root-roles.png)

We have 5 [predefined roles](/reference/rbac#predefined-roles) within our RBAC framework at Unleash.

_Root roles_:

1. Admin
2. Editor
3. Viewer

_Project roles_:

1. Owner
2. Member

> Assign users with root roles to configure entire systems. These users can perform any operation within the Unleash platform and change permissions for other users when they belong to a specific user group.

Author: The ability to perform all operations is just for admin roles.

Editors can do everything **except** managing users and roles. Editors can operate as project owners.

Viewers can operate as project members when added to a project. They can create, view, and update feature flags within a specific project.

Assign other users to project roles to have domain-specific control within a specific project maintained in Unleash.

> Project permissions are separated from root permissions to make it even more targeted regarding what permissions someone can and cannot have for each piece of Unleash. Assign developers with creation and modification rights and viewers who can observe but not change.

Reviewer: "for each piece of Unleash" is too vague. Can we be more specific?

> For more fine-tuned access controls, create [custom root roles](/how-to/how-to-create-and-assign-custom-root-roles) and [custom project roles](/how-to/how-to-create-and-assign-custom-project-roles), where you can define the privileges and limitations beyond the predefined roles we have built into Unleash.

Reviewer: Can we give some examples of the types of access RBAC can be used for here?  I think there are something like 50+ things you can define. We could add a screenshot of that.

### Extend RBAC with a change management workflow

While RBAC allows you to administer Unleash safely, you might need approval processes when changing feature flags or their configuration. When multiple teams are working on complex systems, [change requests](/reference/change-requests) provide a systematic approach to:

-   **Comprehensive review**: Every proposed feature flag modification goes through a review, reducing the likelihood of unintended consequences.
-   **Audit trail**: Every change is documented, timestamped, and attributed to specific team members, creating a permanent record of system modifications.

> -   **Four-eyes approval workflows**: Multi-stage approval processes ensure critical changes are thoroughly reviewed and approved by at least 2 other people before implementation.

Reviewer: Is it really that they are "approved by at least 2 **other** people before implementation."?  That would be 6 eyes. ;). I think, but am not sure, that you can have the requestor and a single approver.  Please confirm.

Author: Definitions online say "at least 2 people need to approve" and yes, it is vague as to whether or not the requestor can be considered one of the "pairs of eyes".

However, looking a second time (post-holiday freshness :) ), I see that the [Open Risk Manual](https://www.openriskmanual.org/wiki/Four_Eyes_Principle) implies that 1 person needs to review, making it a total of 2 people involved.

> -   **Compliance requirements**: For regulated industries like finance and healthcare, Change Requests provide the detailed documentation necessary to meet strict compliance standards.

Reviewer: Let's link here to our new compliance docs section. https://docs.getunleash.io/using-unleash/compliance/compliance-overview

> ![Change requests are divided between two tabs: open and closed change request lists with relevant metadata listed per request.](/img/use-case-user-mgmt-change-requests.png)

Reviewer: This sentence is hard to parse. Can you revisit it please?

> Imagine a large banking application where a development team wants to modify a feature flag controlling a new authentication method. Instead of a developer making an immediate change, the change request workflow might require:

Reviewer: ```suggestion
Imagine a developer working at our large banking platform wants to modify a feature flag controlling a new authentication method. Instead of a developer making an immediate change, the change request workflow might require:
```

1. Initial proposal submission
2. Security team review
3. Compliance officer approval
4. Final sign-off from technical leadership

> ![This GIF shows how to quickly make changes to your flag, request the change, approve it, and apply the changes.](/img/use-case-user-mgmt-cr.gif)

Reviewer: ```suggestion
![This GIF shows how to quickly make changes to your feature flag, request the change, approve it, and apply the changes.](/img/use-case-user-mgmt-cr.gif)
```

> 

Reviewer: I would like to add a link to Mateusz's blog on this topic. https://www.getunleash.io/blog/feature-flag-change-requests-how-to

Something like "For more detail on what enterprise-level change requests should include [read our blog](https://www.getunleash.io/blog/feature-flag-change-requests-how-to)

Unleash supports multiple log export formats, including JSON and CSV, for integration with enterprise reporting and analytics tools. This flexibility allows security teams to transform raw audit logs into meaningful insights about system usage, user behaviors, and potential security risks.

Here’s an example of an audit log entry in JSON format:

```json
{
    "timestamp": "2024-03-15T14:22:11Z",
    "user": {
        "id": "user-123",
        "email": "jane.developer@company.com"
    },
    "action": "feature_flag_updated",
    "details": {
        "flag_name": "payment-gateway-experiment",
        "previous_state": "enabled",
        "new_state": "disabled",
        "environment": "production"
    },
    "source_ip": "192.168.1.100"
}
```

The most advanced implementations integrate Unleash audit logs directly into broader Security Information and Event Management (SIEM) systems. This enables real-time monitoring and creates a holistic view of system interactions across multiple platforms. By correlating Unleash audit data with other enterprise security logs, organizations can detect subtle patterns and potential security anomalies that might go unnoticed in isolated systems.

> By implementing intelligent, automated, and granular access controls, organizations can create a secure foundation for continuous delivery at scale.

Reviewer: This conclusion sentence seems out of place.  Only refers to access controls, not the rest of the topics in the blog.
